subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -481,37 +481,42 @@ By default, the Java compiler runs in the Gradle process. Setting `options.fork`
 
 Starting with Gradle 2.1, it is possible to compile Java incrementally. See the api:org.gradle.api.tasks.compile.JavaCompile[] task for information on how to enable it.
 
-Main goals for incremental compilations are:
+The main goals for incremental compilations are:
 
-* Avoid wasting time compiling source classes that don't have to be compiled. This means faster builds, especially when a change to a source class or a jar does not incur recompilation of many source classes that depend on the changed input.
+* Make incremental builds much faster.
```

Comment:
> Also, doesn't this make all builds that have Java compilation faster? I.e. isn't this feature independent of incremental build?

No. Incremental build tells us what has changed. Without that we couldn't incrementally compile.

Reply:
OK, good to know. I'm now wondering about the term "incremental builds" and how it might be interpreted. Most builds are incremental, right? Particularly those that use the Java Plugin. It seems like you're equating an incremental build to a non-clean build. Is that a fair assessment?

I wonder whether it would be a good idea to distinguish between a build that has up-to-date checks (what I view as an "incremental build") and builds that don't start from the clean state. Or perhaps in this specific case, we can talk about any build that has previously performed the compilation.

I think for now it's probably OK to leave it as it is, but at some point I'd like to standardise terminology around the distinction I've described.

Reply:
> It seems like you're equating an incremental build to a non-clean build. Is that a fair assessment?

That's what an incremental build is.

> a build that has up-to-date checks (what I view as an "incremental build")

Every build has up-to-date checks.

Reply:
> Every build has up-to-date checks.

But the up-to-date checks only do something if the relevant tasks are designed that way. All core Gradle tasks properly support incremental build, but if you don't use the appropriate annotations on your own tasks (or effectively use the dynamic API), you end up with something that's _not_ an incremental build in my book.

Anyway, if you're sure that's the widely accepted and understood meaning of "incremental build" I'll go with it.